### PR TITLE
fix(schematics): normalize path in findTargetProject

### PR DESCRIPTION
**Pull request by luchsamapparat**
findTargetProject returns OS-specific paths which may not match the paths in projectNode.files. To
fix this, the path needs to be normalized the same way as it is done in
isRelativeImportIntoAnotherProject. Also, per default when an import points to a directory, "index"
should be added as a potential import file candidate.

## Current Behavior (This is the behavior we have today, before the PR is merged)
VS Code sometimes auto-generates import paths like this:

`import { ProductsModule } from '../../../products/src';`

However, `nx-enforce-module-boundaries` does not mark this as an error, even though `products` in this example is another library.

## Expected Behavior (This is the new behavior we can expect after the PR is merged)
The linter should throw an error like this:

`library imports must start with @ngxp/ (nx-enforce-module-boundaries)`

## Issue
I found two issues:

1) `findTargetProject` creates a path that is OS-specific. On Windows this means that it contains `\` instead of `/`. That's why `findProjectUsingFile` returns no match, as the file paths in `projectNodes` contain UNIX-style paths.

2) The path generated by VS Code does already contain the `/src` part that was added in `findTargetProject`: `path.join(targetFile, 'src', 'index')`, so the generated path contained `src/src/`. I don't know why this change was made - in an older version of this method, the `src` was missing. I added another case where `index` is added without `src`, however maybe the `src` variant can be removed.

### CODE REVIEW

**Comment by luchsamapparat** [comment]
I'm not sure why the travis build fails 🤔 I couldn't get the build working on my Windows machine, but the build ran successfully on my travis: https://travis-ci.org/luchsamapparat/nx

File: packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts [hunk 1]
```diff
@@ -340,10 +340,10 @@ describe('Enforce Module Boundaries', () => {
             tags: [],
             implicitDependencies: [],
             architect: {},
-            files: [`libs/mylib/src/main.ts`, `libs/mylib/other/index.ts`],
+            files: [`libs/mylib/src/main.ts`, `libs/mylib/src/other/index.ts`],
```
Comment (vsavkin) [inline packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts]: Why is this change necessary? Isn't the code below makes it work without src too?

**Review by vsavkin — CHANGES_REQUESTED** [review]

**Comment by vsavkin** [comment]
@luchsamapparat thank you for your PR! Left a comment.

File: packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts (hunk 1 above)
Comment (luchsamapparat) [reply packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts]: To be honest, I don't know why I changed that test 🤨 I reverted that change and instead added an explicit test case for my change.

But that highlights another thing, that I was already wondering about, when I made the change. Under what circumstances do we have a situation as described in the `should error when relatively importing another library` test case? e.g. an import to `/other` that resolves to `/other/src/index.ts`. When I create a new Nx Workspace, that wouldn't even work with the tsconfig as it is generated. And that begs the question, why this is even part of the existing implementation of the rule:

https://github.com/nrwl/nx/blob/c5fdf1c6562c628387f1efa57de6d56638d7d336/packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.ts#L289

**Comment by luchsamapparat** [comment]
thanks, I updated the commit and brought up another related question...

btw, what is the etiquette here? should I just *resolve* the conversation of your code comment or do I need to post here as well in order to get you notified...?

File: packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts [hunk 2]
```diff
@@ -343,7 +343,7 @@ describe('Enforce Module Boundaries', () => {
             files: [`libs/mylib/src/main.ts`, `libs/mylib/other/index.ts`],
             fileMTimes: {
               'libs/mylib/src/main.ts': 1,
-              'libs/mylib/other/index.ts': 1
+              'libs/mylib/src/other/index.ts': 1
```
Comment (vsavkin) [inline packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts]: this should be reverted too, right?

File: packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts (hunk 1 above)
Comment (vsavkin) [reply packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.spec.ts]: You are right. The TS compiler will fail, so this doesn't seem to be necessary. 

I think that we should have just checked that you aren't importing outside your library. It shouldn't even matter if your target file is in a library or not. It's a bit more restricting but simpler to understand.

**Comment by vsavkin** [comment]
@luchsamapparat just resolving issues is fine. You don't have to add a comment.

I'll merge the PR as is because it fixes a serious issue. But for Nx 8 we should consider refactoring the check (see my comment above)
